nntrainer/tensor/manager.cpp

```diff
@@ -377,28 +375,39 @@ void Manager::deallocateGradients() {
  * @brief Initialize the weight gradients
  */
 void Manager::initializeGradients() {
-  if (total_weight_size == 0) {
-    ml_logw("Nothing done on initialize because there is no weight registered");
-    return;
-  }
-
-  AllocFunc allocate_grad = getAllocFunc(false);
+  if (LAYER_V2) {
+    for (auto &l_w : weights_v2) {
+      for (auto &w : l_w) {
+        w->initializeGradient();
+        // auto exec_order = tensor_exec_order[w.getName()];
+        // tensor_map(&w->getGradientRef(), requestMemory(w.getDim().size(),
+        //       std::get<1>(exec_order), std::get<2>(exec_order) + 1));
+      }
+    }
+  } else {
+    if (total_weight_size == 0) {
```

Comment:
Is this code for the layer v1 tests?

If it is, could you run a test with layer v1, if this is working fine before merging this PR?

Reply:
total_weight_size is used by layer v1 (where requestWeights() interface was used).
As this variable was not needed in layerv2 style, it is just moved in the else case. So, there is no change in behavior for layer-v1 and layer-v2 by bringing this in.

There is no layerv1 tests which directly deals with this.

Reply:
I wonder the reason for the maintaining this section rather then deleting it.

Reply:
It is kept to able to use this strategy in a manual strategy if the automated strategy is not as good (although I am quite sure that automated will do better than manual but I cant guarantee).